File: communication/cpp/test_server.cpp
#include "tcp_protocol_server.h"
#include <string>
#include <iostream>
using namespace std;

const string host = "127.0.0.1";
const int port = 9999;

int main()
{
    tcp_server tccp = tcp_server(host, port);

    tccp.Send<int>(60207);

    int i = tccp.Recv<int>();
    cout << "recv : " << i << endl;


    tccp.Send<double>(60.207);

    double f = tccp.Recv<double>();
    cout << "recv : " << f << endl;

    
    tccp.Send<string>("60-207");

    string s = tccp.Recv<string>();
    cout << "recv : " << s << endl;
}

File: communication/go/test_server/test_server.go
// The folder with go.mod must have the file "tcp_protocol_client.go" with the same package name
package main

import (
	"fmt"
	"tcp/tps"
)

const host = "localhost"
const port = "9999"

func main() {
	conn := tps.InitTCP(host, port)

	var i int64 = 60207
	tps.Send(conn, i)
	_, data := tps.Recv(conn)
	fmt.Print("recv : ")
	fmt.Println(data)

	var f float64 = 60.207
	tps.Send(conn, f)
	_, data = tps.Recv(conn)
	fmt.Print("recv : ")
	fmt.Println(data)

	var l string = "60-207"
	tps.Send(conn, l)
	_, data = tps.Recv(conn)
	fmt.Print("recv : ")
	fmt.Println(data)

	tps.ExitTCP(conn)
}
